examples/style_sheets/style_sheets_reference.py

```diff
@@ -0,0 +1,128 @@
+"""
+This example demonstrates different available style sheets on a common example.
+
+The different plots are heavily similar to the other ones in the style sheet
+gallery.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def plot_scatter(ax, prng, nb_samples=200):
+    """ Scatter plot.
+
+    NB: `plt.scatter` doesn't use default colors.
+    """
+    x, y = prng.normal(size=(2, nb_samples))
+    ax.plot(x, y, 'o')
+    ax.set_xlabel('X-label')
+    ax.set_ylabel('Y-label')
+    return ax
+
+
+def plot_colored_sinusoidal_lines(ax):
+    """ Plot sinusoidal lines with colors from default color cycle.
+    """
+    L = 2*np.pi
+    x = np.linspace(0, L)
+    ncolors = len(plt.rcParams['axes.color_cycle'])
+    shift = np.linspace(0, L, ncolors, endpoint=False)
+    for s in shift:
+        ax.plot(x, np.sin(x + s), '-')
+    ax.margins(0)
+    return ax
+
+
+def plot_bar_graphs(ax, prng, min_value=5, max_value=25, nb_samples=5):
+    """ Plot two bar graphs side by side, with letters as xticklabels.
+    """
+    x = np.arange(nb_samples)
+    ya, yb = prng.randint(min_value, max_value, size=(2, nb_samples))
+    width = 0.25
+    ax.bar(x, ya, width)
+    ax.bar(x + width, yb, width, color=plt.rcParams['axes.color_cycle'][2])
+    ax.set_xticks(x + width)
+    ax.set_xticklabels(['a', 'b', 'c', 'd', 'e'])
+    return ax
+
+
+def plot_colored_circles(ax, prng, nb_samples=15):
+    """ Plot circle patches.
+
+    NB: drawing a fixed amount of samples, rather than using the length of
+    the color cycle, because different styles may have different numbers of
+    colors.
+    """
+    list_of_colors = plt.rcParams['axes.prop_cycle'].by_key()['color']
+    max_idx = min(nb_samples, len(list_of_colors))
+    for color in list_of_colors[0:max_idx]:
```

Comment:
I think something like this would work (which granted exercises more than color, but that seems like a feature).

```python
for sty, j in zip(plt.rcParams['axes.prop_cycle'], range(nb_samples)):
    ax.add_patch(Circle(prng.normal(scale=3, size=2),
                      radius=1.0, **sty))
```

`Cyclers` are iterables that yield dictionaries exactly for `**`ing into functions like this.